### ISSUE TITLE
Gateway Asset Amount Rounding

### ISSUE THREAD

**Issue by ghost** [omgnetwork/optimism#533]
The Gateway Asset Amount currently rounds to the hundredths. Some assets like ETH need more specificity, and rounding to the thousandths or ten thousandths.

For example, I fast bridge .10 ETH. I ended up receiving .0999 ETH.

The Gateway shows that I have .10 ETH:
![image](https://user-images.githubusercontent.com/90710496/134760269-8f4853c3-6eb8-41be-8b6a-417af4c46eb5.png)

When I try to bridge to L2, it tells me the max amount is .100 ETH, but when I enter .1 ETH, it tells me it has to be less than .100 ETH.
![image](https://user-images.githubusercontent.com/90710496/134760351-5c66cce8-3943-4e17-81fa-babd51bc5f5d.png)

It is a bit of nuisance having to look at a separate tab or window to determine the exact amount I have.

**Comment by CAPtheorem** [comment]
Yes, agreed, the rounding is on our radar since it also affects "USE ALL" functionality (Nothing more annoying that having 0.0000000454304 ETH in a wallet)
